### ISSUE TITLE
jsf.js isn't always rendered in the output .jsf page

### ISSUE THREAD

**Issue by csabee** [ButterFaces/ButterFaces#55]
The following bug is making me tear my hair out.
Issue: using autoComplete on the simples form may fail, because butterfaces didn't render jsf.js script to the page.

AS: wildfly 10.
Butterfaces: 2.1.4

Here is the simplest form of it:

JSF page:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:b="http://butterfaces.org/components">
<h:head></h:head>
<body>
	<div class="container">
		<h:form id="intid">
			<div class="col-lg-4" style="margin-top: 5em">
				<b:text value="#{tester.inp}" label="1" id="autoin">
					<b:autoComplete autoComplete="#{tester}" />
				</b:text>
			</div>
		</h:form>
	</div>
</body>
</html>
```

Backing bean:
```
@Model
@Named("tester")
public class Testview implements Serializable, AutoCompleteModel {

	private static final long serialVersionUID = 3198625083156776259L;
	private String inp;

	public String getInp() {
		return inp;
	}

	public void setInp(String inp) {
		this.inp = inp;
	}
	
	public void save(){
		System.out.println("inp: " + inp);
	}
	@Override
	public List<String> autoComplete(Object arg0) {
		List<String> options = new ArrayList<>();
		options.add("aaa");
		options.add("bbb");
		options.add("abab");
		return options;
	}
}
```

**Comment by larmic** [comment]
You are right. At this time you need an ajax-tag to activate this feature

Some cases in ButterFaces required an ajax-tag to activate features. Maybe we should describe it better in our showcase.

This works for me. Is it ok for you?

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:b="http://butterfaces.org/components">
<h:head></h:head>
<body>
<div class="container">
    <h:form id="intid">
        <div class="col-lg-4" style="margin-top: 5em">
            <b:text value="#{autoCompleteShowcase.value}" label="1" id="autoin">
                <f:ajax />
                <b:autoComplete autoComplete="#{autoCompleteShowcase.autoCompleteModel}" />
            </b:text>
        </div>
    </h:form>
</div>
</body>
</html>
```

**Comment by csabee** [comment]
Hi, 

I would think, that both solutions are ok:
1. Extend the documentation (maybe I can help with it?)
2. Fix the issue with the rendering of jsf.js (for future purposes, this would be better).

Thanks

**Comment by larmic** [comment]
You are welcome to support ButterFaces. I would be glad. :)

I will try to add jsf.js. Maybe it is working out-of-the-box.

**Comment by larmic** [comment]
Ok, I've fixed the problem and will build 2.1.6.

Thanks for reporting this bug!

**Comment by larmic** [comment]
@csabee ButterFaces 2.1.6 has been released!
